lighthouse-core/lib/dependency-graph/simulator/network-analyzer.js

```diff
@@ -452,6 +452,19 @@ class NetworkAnalyzer {
     // The main document is the earliest document request, using position in networkRecords array to break ties.
     return documentRequests.reduce((min, r) => (r.startTime < min.startTime ? r : min));
   }
+
+  /**
+   * Resolves redirect chain given a main document.
+   * See: {@link NetworkAnalyzer.findMainDocument}) for how to retrieve main document.
+   *
+   * @param {LH.Artifacts.NetworkRequest|undefined} request
+   * @returns {LH.Artifacts.NetworkRequest|undefined}
+   */
+  static resolveRedirects(request) {
+    if (!request) return undefined;
```

Comment:
Can we handled the undefined case where this is called instead?

Reply:
I'm good with either, but this came from [this suggestion](https://github.com/GoogleChrome/lighthouse/pull/11620#discussion_r516281777)

cc @patrickhulce

Reply:
I suggested that so it would be drop-in replacement with the rest :)

I would also have a slight preference to handle the undefined at the caller, but I don't think it matters much honestly.

Reply:
Alrighty, I leave it to you @bradfrosty, we're good with w/e!

Reply:
Moved the undefined case to the caller -- let me know if that looks alright! Instead of handling the possible exception (like with the call to `findMainDocument()`), I type checked the `mainDocument` .

I also updated the undefined case test in `network-analyzer-test.js` to expect an exception to throw.